lib/cldr/export/data/units.rb

```diff
@@ -44,6 +50,22 @@ def duration_unit
             result[node.attribute("type").value.to_sym] = node.xpath("durationUnitPattern").first.content
           end
         end
+
+        def units_xpath_to_key(xpath)
+          match = xpath.match(%r{^\.\./unitLength\[@type='([^']*)'\]$})
+          raise StandardError, "Didn't find expected data in alias path attribute: #{xpath}" unless match
+
+          type = match[1]
+          :"units.unitLength.#{type}"
+        end
+
+        def unit_xpath_to_key(xpath)
+          match = xpath.match(%r{^\.\./unit\[@type='([^']*)'\]$})
+          raise StandardError, "Didn't find expected data in alias path attribute: #{xpath}" unless match
+
+          type = match[1]
+          :"units.unitLength.short.#{type}"
```

Comment:
```suggestion
          length = match[1]
          :"units.unitLength.#{length}.#{unit_type}"
```

Have you tested this one to show that it successfully finds the right value?

Reply:
Nevermind. I was reading the wrong code. Sorry.